lib/utils/tty.ts

```diff
@@ -0,0 +1,72 @@
+import * as Bluebird from 'bluebird';
+
+const windowSize: { width?: number; height?: number } = {};
+
+const updateWindowSize = () => {
+	const size = require('window-size')?.get();
+	windowSize.width = size?.width;
+	return (windowSize.height = size?.height);
+};
+
+process.stdout.on('resize', updateWindowSize);
+
+export = (stream: NodeJS.WriteStream = process.stdout) => {
+	// make sure we get initial metrics
+	updateWindowSize();
+
+	const currentWindowSize = () => {
+		// always return a copy.
+		// width/height can be undefined if no TTY.
+		return {
+			width: windowSize.width,
+			height: windowSize.height,
+		};
+	};
+
+	const hideCursor = () => stream.write('\u001B[?25l');
+
+	const showCursor = () => stream.write('\u001B[?25h');
+
+	const cursorUp = (rows: number = 0) => {
+		return stream.write(`\u001B[${rows}A`);
+	};
+
+	const cursorDown = (rows: number = 0) => {
+		return stream.write(`\u001B[${rows}B`);
```

Comment:
NIT: `cursorUp` and `cursorDown` could be written in the same style as the other functions, without the `return` keyword.